contracts/base/Dispatcher.sol

```diff
@@ -33,171 +36,199 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks,
+                        }
+                        Payments.unwrapWETH9(map(recipient), amountMin);
+                    } else if (command == Commands.PERMIT2_TRANSFER_FROM_BATCH) {
+                        (IAllowanceTransfer.AllowanceTransferDetails[] memory batchDetails) =
+                            abi.decode(inputs, (IAllowanceTransfer.AllowanceTransferDetails[]));
+                        permit2TransferFrom(batchDetails, lockedBy);
+                    } else if (command == Commands.BALANCE_CHECK_ERC20) {
+                        // equivalent: abi.decode(inputs, (address, address, uint256))
+                        address owner;
+                        address token;
+                        uint256 minBalance;
+                        assembly {
+                            owner := calldataload(inputs.offset)
+                            token := calldataload(add(inputs.offset, 0x20))
+                            minBalance := calldataload(add(inputs.offset, 0x40))
+                        }
+                        success = (ERC20(token).balanceOf(owner) >= minBalance);
+                        if (!success) output = abi.encodePacked(BalanceTooLow.selector);
+                    } else {
+                        // placeholder area for command 0x0f
+                        revert InvalidCommandType(command);
+                    }
                 }
-                // 0x08 <= command < 0x10
+                // 0x10 <= command < 0x18
             } else {
-                if (command == Commands.V2_SWAP_EXACT_IN) {
-                    // equivalent: abi.decode(inputs, (address, uint256, uint256, bytes, bool))
-                    address recipient;
-                    uint256 amountIn;
-                    uint256 amountOutMin;
-                    bool payerIsUser;
-                    assembly {
-                        recipient := calldataload(inputs.offset)
-                        amountIn := calldataload(add(inputs.offset, 0x20))
-                        amountOutMin := calldataload(add(inputs.offset, 0x40))
-                        // 0x60 offset is the path, decoded below
-                        payerIsUser := calldataload(add(inputs.offset, 0x80))
-                    }
-                    address[] calldata path = inputs.toAddressArray(3);
-                    address payer = payerIsUser ? lockedBy : address(this);
-                    v2SwapExactInput(map(recipient), amountIn, amountOutMin, path, payer);
-                } else if (command == Commands.V2_SWAP_EXACT_OUT) {
-                    // equivalent: abi.decode(inputs, (address, uint256, uint256, bytes, bool))
-                    address recipient;
-                    uint256 amountOut;
-                    uint256 amountInMax;
-                    bool payerIsUser;
+                if (command == Commands.ERC721_PERMIT) {
+                    (success, output) = address(V3_POSITION_MANAGER).call(inputs);
```

Comment:
this needs to check its the `permit` selector 👀 
currently you could call `decreaseLiq` on a token you arent meant to access

Reply:
would be great to have a test that reverts passing another selector to permit too

Reply:
✅